### ISSUE TITLE
Cache the network latency of the subgroup nodes to improve the speed of requests

### ISSUE THREAD

**Issue by bakhshandeh** [muon-protocol/muon-node-js#63]
When requests run on Muon, the network latency of the nodes that participate in the signature is important and the speed depends on the slowest connection between the nodes. Here are the times in milliseconds for a request:

```json
{  
    "find fastest online nodes": 2886,  
    "req exec time": 2,  
   "dkg time": 1281,  
   "req broadcast": 42,  
   "confirm waiting": 346  
}
```

The time it takes to find online nodes can be considerable. One solution to improve speed is to monitor the network latency between all of the subgroup nodes and cache it on the nodes for some time. This way, the gateway node that wants to initiate the request knows the status of all nodes and can select the best group to run the request.

**Comment by bakhshandeh** [comment]
**How to implement?**

1- Add a new plugin(`latency-check`) to check the latency between the current node and all of the nodes that are in the same party with it and cache in the memory

2- Update `__getAppPartyLatency` to use cached the latency of nodes from the cache instead of calling the nodes in realtime

3- Gateway of the request should be updated to exclude the nodes that have not replied to gateway. It helps the gateway to remove the nodes that are offline. But the latency of them are valid on the cache

**Comment by sadeghte** [comment]
implemented in #142
